internal/c/parts/audio/audio.cpp

```diff
@@ -2654,21 +3013,50 @@ int32_t func__sndopen(qbs *qbsFileName, qbs *qbsRequirements, int32_t passed) {
             std::string_view(reinterpret_cast<const char *>(qbsFileName->chr), qbsFileName->len));                        // make a unique key and save it
         audioEngine.bufferMap.AddBuffer(qbsFileName->chr, qbsFileName->len, audioEngine.soundHandles[handle]->bufferKey); // make a copy of the buffer
         auto [buffer, bufferSize] = audioEngine.bufferMap.GetBuffer(audioEngine.soundHandles[handle]->bufferKey);         // get the buffer pointer and size
-        audioEngine.maResult = audioEngine.InitializeSoundFromMemory(buffer, bufferSize, handle);                         // create the ma_sound
+        auto fname = std::to_string(audioEngine.soundHandles[handle]->bufferKey);                                         // convert the buffer key to a string
+
+        audioEngine.maResult = ma_sound_init_from_file(&audioEngine.maEngine, fname.c_str(), audioEngine.soundHandles[handle]->maFlags, NULL, NULL,
+                                                       &audioEngine.soundHandles[handle]->maSound); // create the ma_sound
     } else {
         std::string fileName(reinterpret_cast<char const *>(qbsFileName->chr), qbsFileName->len);
 
-        AUDIO_DEBUG_PRINT("Loading sound from file '%s'", fileName.c_str());
+        if (audioEngine.soundHandles[handle]->maFlags & MA_SOUND_FLAG_STREAM) {
+            AUDIO_DEBUG_PRINT("Streaming sound from file '%s'", fileName.c_str());
 
-        // Forward the request to miniaudio to open the sound file
-        audioEngine.maResult = ma_sound_init_from_file(&audioEngine.maEngine, filepath_fix_directory(fileName), audioEngine.soundHandles[handle]->maFlags, NULL,
-                                                       NULL, &audioEngine.soundHandles[handle]->maSound);
+            audioEngine.maResult = ma_sound_init_from_file(&audioEngine.maEngine, fileName.c_str(), audioEngine.soundHandles[handle]->maFlags, NULL, NULL,
+                                                           &audioEngine.soundHandles[handle]->maSound); // create the ma_sound
+        } else {
+            AUDIO_DEBUG_PRINT("Loading sound from file '%s'", fileName.c_str());
+
+            auto contents = AudioEngine_LoadFile<std::string>(fileName.c_str());
+
+            if (contents.empty()) {
+                AUDIO_DEBUG_PRINT("Failed to open sound file '%s'", fileName.c_str());
+
+                audioEngine.soundHandles[handle]->isUsed = false;
+
+                return AudioEngine::INVALID_SOUND_HANDLE;
+            }
+
+            AUDIO_DEBUG_PRINT("Sound length: %zd", contents.length());
+
+            // Configure a miniaudio decoder to load the sound from memory
+            audioEngine.soundHandles[handle]->bufferKey = std::hash<std::string_view>{}(contents);                            // make a unique key and save it
+            audioEngine.bufferMap.AddBuffer(contents.data(), contents.length(), audioEngine.soundHandles[handle]->bufferKey); // make a copy of the buffer
```

Comment:
IMO it would be worth avoiding the copy of the file contents when putting the buffer into the BufferMap, that was something I disliked about my simple solution.

Reply:
Good idea. Done.